src/Relation/Nullary/Decidable.agda

```diff
@@ -80,3 +80,10 @@ dec-yes-irr a? irr a with a′ , eq ← dec-yes a? a rewrite irr a a′ = eq
 
 ⌊⌋-map′ : ∀ t f (a? : Dec A) → ⌊ map′ {B = B} t f a? ⌋ ≡ ⌊ a? ⌋
 ⌊⌋-map′ t f a? = trans (isYes≗does (map′ t f a?)) (sym (isYes≗does a?))
+
+does-≡ : (a? b? : Dec A) → does a? ≡ does b?
+does-≡ a? (yes a) = dec-true a? a
+does-≡ a? (no ¬a) = dec-false a? ¬a
```

Comment:
I would always be tempted, where possible, to make such definitions lazier wrt their `Dec A` arguments, as follows:
```suggestion
does-≡ a? (true because [a]) = dec-true a? (invert [a])
does-≡ a? (false because [¬a]) = dec-false a? (invert [¬a])
```
because, in general, `yes` and `no` are unnecessarily strict as constructors.

Reply:
hmm, I see. your concern is that `yes` / `no` makes the function strict on the Reflects, right? I can see the advantage of your proposed code, but the downside is making the proof more inscrutable to the casual reader... and bearing in mind this is computing an identity, not a "useful" value as in `Data.List.Base`, I'm not sure this extra lazyness matters here?

(I wish Agda had lazy patterns like Haskell... :pensive:)

Reply:
It's a fair point... so I'm happy to give way on this one. But long term I would like to refactor `Dec` to avoid `yes` and `no` entirely in favour of such lazier counterparts... something I began with #2055 and subsequent PRs such as #2149 

But the recent issue #2415 and its retrospective consequences for #2359 / #2365 suggest that a bit more thinking is (still) required. 

Thanks for at least considering the suggestion. And as for the 'casual' reader of library source, there are certainly interesting trade-offs to be made between such an audience, and that of developers/maintainers... as I have learnt over the last 3 years ;-)

Reply:
yeah, there's certainly a tradeoff... I feel like I could be convinced either way tbh